react/components/ComponentEditor/index.tsx

```diff
@@ -523,6 +630,50 @@ class ComponentEditor extends Component<
     }
   }
 
+  private handleConfigurationDelete = async () => {
+    const configurationId = this.state.deletedConfigurationId
+    const { deleteConfiguration, editor, runtime } = this.props
+
+    this.setState({
+      isLoading: true,
+    })
+
+    try {
+      await deleteConfiguration({
+        variables: {
+          configurationId,
+          extensionName: editor.editTreePath,
+          routeId: runtime.page,
+        },
+      })
+
+      const extensionConfigurationsQuery = this.props.extensionConfigurations
+
+      await extensionConfigurationsQuery.refetch({
+        configurationsIds:
+          runtime.extensions[editor.editTreePath as string].configurationsIds,
+        treePath: editor.editTreePath,
+      })
+
+      this.setState(
+        {
+          isLoading: false,
+        },
+        () => {
+          this.handleModalClose()
+        },
+      )
+      this.setState({isDeleteModalOpen: false,})
+      if(this.props.extensionConfigurations.extensionConfigurations.length === 0){ this.setState({isEditMode: true,})}
+      else{ this.handleConfigurationChange(this.props.extensionConfigurations.extensionConfigurations[0])}
+
+    } catch (err) {
+      alert('Something went wrong. Please try again.')
```

Comment:
This is a really bad way to give user feedback...

Reply:
i just copied the error message from handleSaveConfiguration method. what do you think i should put there instead?

Reply:
Let's put this on hold until I finish the PR for the [https://styleguide.vtex.com/#/Components/Notification/ToastProvider](Toast component)!